include/zephyr/posix/pthread.h

```diff
@@ -7,14 +7,15 @@
 #ifndef ZEPHYR_INCLUDE_POSIX_PTHREAD_H_
 #define ZEPHYR_INCLUDE_POSIX_PTHREAD_H_
 
+#include "pthread_key.h"
```

Comment:
Probably shouldn't use " path here -- this is a public header and we don't want to accidentally pull in an application -specific header?

Reply:
It's not _really_ a public header - and what I mean by that is that it's a not a standard posix header.

In any case, it's not being added, removed or modified in this PR.